x-pack/filebeat/input/httpjson/request.go

```diff
@@ -163,21 +181,106 @@ func newRequester(
 	}
 }
 
-func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
-	req, err := r.requestFactory.newHTTPRequest(stdCtx, trCtx)
+// collectResponse returns response from provided request
+func (rf *requestFactory) collectResponse(stdCtx context.Context, trCtx *transformContext, r *requester, publisher inputcursor.Publisher) (*http.Response, error) {
+	req, err := rf.newHTTPRequest(stdCtx, trCtx)
 	if err != nil {
-		return fmt.Errorf("failed to create http request: %w", err)
+		return nil, fmt.Errorf("failed to create http request: %w", err)
 	}
-
 	httpResp, err := r.client.do(stdCtx, trCtx, req)
 	if err != nil {
-		return fmt.Errorf("failed to execute http client.Do: %w", err)
+		return nil, fmt.Errorf("failed to execute http client.Do: %w", err)
+	}
+	return httpResp, nil
+}
+
+// generateNewUrl returns new url value using replacement from oldUrl with str ids
+func generateNewUrl(replacement string, oldUrl string, str []byte) (url.URL, error) {
+	reg, err := regexp.Compile(replacement)
+	if err != nil {
+		return url.URL{}, fmt.Errorf("failed to create regex on provided value: %w", err)
+	}
+	newUrl, err := url.Parse(string(reg.ReplaceAll([]byte(oldUrl), str)))
+	if err != nil {
+		return url.URL{}, fmt.Errorf("failed to replace value in url: %w", err)
+	}
+	return *newUrl, nil
+
+}
+
+func (r *requester) doRequest(stdCtx context.Context, trCtx *transformContext, publisher inputcursor.Publisher) error {
```

Comment:
The function `doRequest` is really long now, what do you think about splitting it into smaller chunks and naming and commenting accordingly?

Reply:
Yes i will do that